### PR TITLE
Remove enterprise confirmations

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Removing enterprise confirmations for #946. I think there isn't a specific issue for this, but it's mentioned somewhere in a list of "issues that need to be created".

There were a lot of references to the enterprise confirmations throughout the codebase, so this one might need some fine-tuning. 

Shouldn't be merged before #1690 is finished and merged.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -123,8 +123,8 @@
   create_table "distributors_shipping_methods", :force => true do |t|
     t.integer  "distributor_id"
     t.integer  "shipping_method_id"
-    t.datetime "created_at",         :null => false
-    t.datetime "updated_at",         :null => false
+    t.datetime "created_at"
+    t.datetime "updated_at"
```
Comment (enricostano) [inline db/schema.rb]: Why this change?

File: db/schema.rb [hunk 2]
```diff
@@ -997,8 +992,13 @@
     t.datetime "reset_password_sent_at"
     t.string   "api_key",                :limit => 40
     t.integer  "enterprise_limit",                     :default => 1, :null => false
+    t.string   "confirmation_token"
+    t.datetime "confirmed_at"
+    t.datetime "confirmation_sent_at"
+    t.string   "unconfirmed_email"
```
Comment (enricostano) [inline db/schema.rb]: Are this coming from another PR?

File: db/schema.rb [hunk 3]
```diff
@@ -1089,6 +1090,7 @@
     t.string   "sku"
     t.boolean  "on_demand"
     t.datetime "permission_revoked_at"
+    t.datetime "import_date"
```
Comment (enricostano) [inline db/schema.rb]: ?

File: db/schema.rb (hunk 2 above)
Comment (Matt-Yorkley) [inline db/schema.rb]: That's really weird.  I just added a single migration to remove the enterprise confirmation columns and ran `rake db:migrate`. Maybe I need to fetch from upstream and rebase...

File: lib/open_food_network/users_and_enterprises_report.rb [hunk 4]
```diff
@@ -63,14 +65,25 @@ def users_and_enterprises
     end
 
     def sort(results)
+      pp results
```
Comment (enricostano) [inline lib/open_food_network/users_and_enterprises_report.rb]: ⚠️

Comment (Matt-Yorkley) [inline lib/open_food_network/users_and_enterprises_report.rb]: Yeah I added that in for debugging... still working on the last few failing specs.

File: lib/open_food_network/users_and_enterprises_report.rb [hunk 5]
```diff
@@ -63,14 +65,25 @@ def users_and_enterprises
...
     end
 
     def to_bool(value)
       ActiveRecord::ConnectionAdapters::Column.value_to_boolean(value)
     end
+
+    def to_local_datetime(string)
+      return "" if string.nil?
+      string.to_datetime.in_time_zone.strftime "%Y-%m-%d %H:%M"
+    end
```
Comment (enricostano) [inline lib/open_food_network/users_and_enterprises_report.rb]: Can we use a more descriptive var name than `string`?

Comment (Matt-Yorkley) [inline lib/open_food_network/users_and_enterprises_report.rb]: That was what the original method looked like in `master`, I was just putting it back in. I'll change it now...

File: db/schema.rb (hunk 2 above)
Comment (oeoeaio) [inline db/schema.rb]: @Matt-Yorkley If your dev db contained this migration at the time you ran rake `db:migrate`, I think these would have been added to your schema? I could be wrong but I seem to remember this happening to me before. The way schema.rb gets updated is still a bit of a black box to me...

**Comment by oeoeaio** [comment]
I feel like the transfer of responsibility for checking confirmation needs to switch from `enterprise` to `user` simultaneously, but it is impossible to do that until #1690 is merged though right? I'm just concerned that by removing all of the responsibility from `enterprise` in one step (this one) and then trying to remember to add all of that responsibility back to `user` at a later point is a recipe for trouble. I suppose we just have to be super systematic about it...

File: app/models/enterprise.rb [hunk 6]
```diff
@@ -198,10 +188,6 @@ def self.distinct_count
     count(distinct: true)
   end
 
-  def activated?
-    confirmed_at.present? && sells != 'unspecified'
```
Comment (oeoeaio) [inline app/models/enterprise.rb]: I think we still need to concept of `activated`? It behaves like a check on the enterprise to determine whether the registration process is complete, if anything I think this should just become:

````ruby
owner.confirmed? && sells != 'unspecified'
````

File: app/models/enterprise_relationship.rb [hunk 7]
```diff
@@ -30,7 +30,7 @@ class EnterpriseRelationship < ActiveRecord::Base
   # Load an array of the relatives of each enterprise (ie. any enterprise related to it in
   # either direction). This array is split into distributors and producers, and has the format:
   # {enterprise_id => {distributors: [id, ...], producers: [id, ...]} }
-  def self.relatives(activated_only=false)
+  def self.relatives
```
Comment (oeoeaio) [inline app/models/enterprise_relationship.rb]: Ditto with `activated`

File: app/models/enterprise.rb (hunk 6 above)
Comment (Matt-Yorkley) [inline app/models/enterprise.rb]: @oeoeaio I considered that as well, but won't all owners be confirmed if they have to confirm their email before registering an enterprise or logging in to the backend?

Comment (Matt-Yorkley) [inline app/models/enterprise.rb]: Is it even possible to create an enterprise without specifying the `sells` attribute?

**Comment by Matt-Yorkley** [comment]
I'll move this PR into another branch to make additional changes at the same time.

File: app/models/enterprise.rb (hunk 6 above)
Comment (oeoeaio) [inline app/models/enterprise.rb]: @Matt-Yorkley, owners of all *new* enterprises will be confirmed, but we have to handle legacy data unfortunately...

**Comment by oeoeaio** [comment]
For reference: superseded by #1720
